lib/active_merchant/billing/gateways/flow.rb

```diff
@@ -24,153 +26,175 @@ def initialize options = {}
+        else
+          Response.new false, 'Flow refund - Error', { response: response }
+        end
+      rescue Io::Flow::V0::HttpClient::ServerError => exception
+        error_response exception
       end
 
       # store credit card with flow and get reference token
       def store credit_card, options={}
-        response = get_flow_cc_token credit_card
+        response = cc_with_token credit_card
         Response.new true, 'Credit card stored', { response: response, token: response.token }
       rescue Io::Flow::V0::HttpClient::ServerError => exception
         error_response exception
       end
 
-      private
-
-      def flow_instance
-        FlowCommerce.instance token: @flow_api_key
-      end
+      # stores credit card and returns credit card Flow token String id
+      def store input
+        credit_card =
+        case input
+          when Hash
+            ActiveMerchant::Billing::CreditCard.new input
+          when ActiveMerchant::Billing::CreditCard
+            input
+          when String
+            return input
+          else
+            raise 'Unsuported store method input type [%s]' % input.class
+        end
 
-      def get_flow_cc_token credit_card
         data = {    number: credit_card.number,
                       name: '%s %s' % [credit_card.first_name, credit_card.last_name],
                        cvv: credit_card.verification_value,
            expiration_year: credit_card.year.to_i,
           expiration_month: credit_card.month.to_i
         }
 
-        card_form = ::Io::Flow::V0::Models::CardForm.new data
-        flow_instance.cards.post @flow_organization, card_form
+        response  = flow_instance.cards.post @flow_organization, data
+        response.token
+      end
+
+      private
+
+      def flow_instance
+        FlowCommerce.instance token: @flow_api_key
       end
 
       def error_response exception_object
-        message = if exception_object.respond_to?(:body) && exception_object.body.length > 0
+        message =
+        if exception_object.is_a?(String)
+          exception_object
+        elsif exception_object.respond_to?(:body) && exception_object.body.length > 0
           description  = JSON.load(exception_object.body)['messages'].to_sentence
           '%s: %s (%s)' % [exception_object.details, description, exception_object.code]
-        else
+        elsif exception_object.respond_to?(:message)
           exception_object.message
+        else
+          raise ArgumentError.new('Unsuported exception_object [%s]' % exception_object.class)
         end
 
+        msg = 'ERROR: %s' % message
+        msg = msg.yellow if msg.respond_to?(:yellow)
+
+        puts msg
+
         Response.new false, message, exception: exception_object
       end
 
       def assert_currency currency, amount
-        raise ArgumentError, 'currency not provided' unless currency
+        unless currency.to_s.length == 3
```

Comment:
suggest we do not hardcode here, instead use `lib-reference-ruby` to validate the currency:
https://github.com/flowcommerce/lib-reference-ruby/blob/master/lib/flowcommerce-reference.rb#L13

Reply:
ok